### ISSUE TITLE
SemVer or continue with 0.1.++ ??

### ISSUE THREAD

**Issue by IsmaelMartinez** [IsmaelMartinez/teams-for-linux#78]
Hi, 

This is a open question. 

It seems like @JammieMagee is not really working on version 0.2.0 and @ivelkov have abandon the project (almost a year since last communication)

For that reason, I think it might be better if you move to some sort or release versioning (other than 0.1.++) but happy with either. I don't want to add extra unnecessary work/overhead, but might be nice to standarise this a bit more.  

Specially directed to @benyanke and @julian-alarcon

**Comment by benyanke** [comment]
I'm a big fan of frequent releases and semver. Makes everything far simpler.

That would be my vote.

**Comment by IsmaelMartinez** [comment]
Maybe we can investigate this as an option:

https://github.com/semantic-release/semantic-release/blob/master/README.md

Automation all the way.

**Comment by IsmaelMartinez** [comment]
I will see if I can get that working at some point. At the moment, lets continue just packing up in master and, when a release is ready, increase the version one point. 

If/when more people starts to contribute, we can revisit the idea.,,,

**Comment by julian-alarcon** [comment]
Ok, talking about numbers (and not the how). I think that this a temporary project, until MS supports Linux.
We don't know when is going to be that. And besides that 0.1.x versions was OK to begin as the missing features and errors were multiple, but right now that's not the case, so, I think that we should start making Semantic versioning and define 1.0 version when CI/CD is on place on all the current platforms. And if there is a similar brake on "API" (e.g. change on User Agent) that requires deletion of profile then we should increase the first digit.

About the how, https://github.com/semantic-release/semantic-release/blob/master/README.md seems great but I don't know if it's worthy to complicate all with more dependencies and stuff. I like the idea but it will take time and I don't see a big problem to be solve.

**Comment by IsmaelMartinez** [comment]
Seems like you two are on board with the semver.

Ok, I also think that probably is also best to pack stuff in releases. This is mainly to avoid making package maintainers job easier (like arch and snap). I don't think people want to be doing builds every few day... it is maybe ok every week, but you don't want people hate us.

Should we move then to 0.2.0 (as there are new features) for this release (instead of 0.1.18)? 

I am happy to then increase the release number depending on what is been release (patch = bugs, minor = features, major = breaking changes). 

Probably we can revert back to tell people can create PRs to Master and we/I can deal with determining the version number.

I agree semantic-release might be a pain.. but I might give it a try as it could, in theory, creates the release notes and all that part for me/us. So I can concentrate on doing more important stuff.

**Comment by benyanke** [comment]
Minor note on the snap - I do expect that to be nearly hands off, as that would be part of CI. Master branch hits the `edge` channel, and releases automatically get pushed to the main channel.

**Comment by IsmaelMartinez** [comment]
Ok, I just updated the main README.md to indicate we are using SemVer but people should not worry to increase the number... I/We can decide before a release what number release it with.

The only thing is that, after a release, I will increase the lower version number. Like now that is 0.1.18. That way, we don't re-release previous releases.

Once we are ready to release, we can either leave the number (if only bug fixes) or increase the minor or mayor depending on the case.

Closing this :) Thanks for the input!
